ngrinder-controller/src/main/java/org/ngrinder/agent/model/NGrinderPackage.java

```diff
@@ -0,0 +1,219 @@
+	 * @return String  module full name.
+	 */
+	private String getDistributionPackageName(String regionName, String connectionIP, String ownerName, boolean forWindow) {
+		return getPackageName() + getFilenameComponent(regionName) + getFilenameComponent(connectionIP) +
+			getFilenameComponent(ownerName) + (forWindow ? ".zip" : ".tar");
+	}
+
+	/**
+	 * Get package name
+	 *
+	 * @return String module full name.
+	 */
+	public String getPackageName() {
+		return this.getModuleName() + "-" + config.getVersion();
+	}
+
+	public static void cleanUpPackageDir(boolean all) {
+		synchronized (AgentPackageService.class) {
+			final File packagesDir = NGrinderPackage.getPackagesDir();
+			final File[] files = packagesDir.listFiles();
+			if (files != null) {
+				for (File each : files) {
+					if (!each.isDirectory()) {
+						long expiryTimestamp = each.lastModified() + (TIME_MILLIS_OF_DAY * 2);
+						if (all || expiryTimestamp < System.currentTimeMillis()) {
+							FileUtils.deleteQuietly(each);
+						}
+					}
+				}
+			}
+		}
+	}
+
+	private String getFilenameComponent(String value) {
+		value = trimToEmpty(value);
+		if (isNotEmpty(value)) {
+			value = "-" + value;
+		}
+		return value;
+	}
+
+	private Set<String> getDependentLibs(URLClassLoader urlClassLoader) {
+		Set<String> libs = new HashSet<>();
+		try (InputStream dependencyStream = urlClassLoader.getResourceAsStream(this.getDependenciesFileName())) {
+			final String dependencies = IOUtils.toString(dependencyStream);
+			for (String each : StringUtils.split(dependencies, ";")) {
+				libs.add(each.trim().replace("-SNAPSHOT", ""));
+			}
+		} catch (Exception e) {
+			LOGGER.error("Error while loading " + this.getDependenciesFileName(), e);
+		}
+		return libs;
+	}
+
+	private Map<String, Object> getAgentConfigParam(String regionName, String controllerIP, int port, String owner) {
+		Map<String, Object> confMap = newHashMap();
+		confMap.put("controllerIP", controllerIP);
+		confMap.put("controllerPort", String.valueOf(port));
+		if (StringUtils.isEmpty(regionName)) {
+			regionName = "NONE";
+		}
+		if (StringUtils.isNotBlank(owner)) {
+			if (StringUtils.isEmpty(regionName)) {
+				regionName = "owned_" + owner;
+			} else {
+				regionName = regionName + "_owned_" + owner;
+			}
+		}
+		confMap.put("controllerRegion", regionName);
+		return confMap;
+	}
+
+	@Component
+	public static class NGrinderPackageConfigBeanInjector {
+		@Autowired
+		private Config config;
+
+		@Autowired
+		private ScheduledTaskService scheduledTaskService;
+
+		@PostConstruct
+		public void postConstruct() {
+			NGrinderPackage.config = config;
+			// clean up package directories not to occupy too much spaces.
+			NGrinderPackage.cleanUpPackageDir(true);
+
+			scheduledTaskService.addFixedDelayedScheduledTask(new Runnable() {
+				@Override
+				public void run() {
+					NGrinderPackage.cleanUpPackageDir(false);
+				}
+			}, TIME_MILLIS_OF_DAY);
+		}
+	}
```

Comment:
Did this cleanup code exist before?

Reply:
Yes, this code exist before.
it cleanup cached package (in {ngrinder-home}/download) when start up server.
the scheduled task is run daily and remove package for more than 2 days old.

Reply:
Is this scheduled task essential? 
the content of package will not change unless new deploy.
so this task seems unnecessary.